src/lxc/conf.c

```diff
@@ -1652,29 +1659,27 @@ static int lxc_setup_devpts(struct lxc_conf *conf)
 		return -1;
 	}
 
-	/* mount new devpts instance */
-	ret = mount("devpts", "/dev/pts", "devpts", MS_NOSUID | MS_NOEXEC, devpts_mntopts);
-	if (ret < 0) {
-		/* try mounting without "max" */
-		if (errno == EINVAL) {
-			devpts_mntopts[sizeof(default_devpts_mntopts) - 1] = '\0';
-			ret = mount("devpts", "/dev/pts", "devpts",
-				    MS_NOSUID | MS_NOEXEC, devpts_mntopts);
-			if (ret < 0) {
-				SYSERROR("Failed to mount new devpts instance");
-				return -1;
-			}
-		}
+	/* gid=5 && max= */
+	mount_options[0] = devpts_mntopts;
+	/* !gid=5 && max= */
+	mount_options[1] = devpts_mntopts + sizeof("gid=5");
+	/* gid=5 && !max= */
+	mount_options[2] = default_devpts_mntopts;
+	/* !gid=5 && !max= */
+	mount_options[2] = default_devpts_mntopts + sizeof("gid=5");
```

Comment:
Should be `[3]`, and then `mount_options` needs to be resized to 5 to not break the loop condition below.

Reply:
Thanks, added!